### DISCUSSION TITLE
Project Reunion 0.5 Preview 🎉🐱‍💻

### DISCUSSION THREAD

**Discussion by dayman40** [microsoft/WindowsAppSDK#552] [Announcements]
We've just released our first preview of the 0.5 release of Project Reunion! This is the first release of Project Reunion that includes WinUI 3. This release consists of stabilization improvements, critical bug fixes, and some new functionality. This is an opportunity for our customers to try this out and provide feedback. This is the first round of validating our work with customers prior to release. We’ll continue to refine and improve the process going forward.

## What's new in Project Reunion 0.5 Preview
- WinUI 3 is now part of Project Reunion!
  - In-app acrylic is now supported!
  - The Pivot control is no longer supported, and has been deprecated in WinUI 3. We recommend using the [NavigationView control](https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/navigationview) for your in-app navigation scenarios.
- Project Reunion will be supported down-level to Windows 10 version 1809 - it requires build 17763 or newer.
- A few bug fixes, including:
  - App does not respond to theme changes in Windows when selecting Windows Default theme until restarted
  - ProgressBar doesn't show difference between Paused and Error option 
  - Crash in Desktop XamlControlsGallery when trying to reorder ListView items with touch

Please keep in mind that Project Reunion 0.5 Preview is still a pre-release product, and has a number of [known issues and limitations](https://docs.microsoft.com/en-us/windows/apps/project-reunion/#limitations-and-known-issues) that our team continues to work on. 

This preview is not intended for production apps.

## Getting started with Project Reunion 0.5 Preview
First, you'll need to set up your dev environment with the appropriate technologies. See our documentation for installation instructions and more information on project types:

- [Install Project Reunion 0.5 Preview](https://docs.microsoft.com/windows/apps/project-reunion/#set-up-your-development-environment)
- [Create WinUI 3 Projects](https://docs.microsoft.com/windows/apps/winui/winui3#create-WinUI-projects)

API Reference documentation for Project Reunion 0.5 Preview is coming soon.

Take a look at the walk-through documentation on [getting started with Project Reunion](https://docs.microsoft.com/en-us/windows/apps/project-reunion/#get-started-developing-with-project-reunion)!

We love feedback! We encourage you to file any bug, big or small, on our repo using this template. Knowing which issues and features are important and/or critical to our customers highly influences which ones we tackle.

## What’s Supported

- This preview is not intended for production apps
- Only MSIX packaged desktop apps (WPF, WinForms, Win32, and console) can use the 0.5 preview
- To use WinUI 3, it's best to start a new app project and, if useful, copy any existing code/business logic from past projects into the new project

The following features are part of the 0.5 preview...
- [WinUI 3 ](https://docs.microsoft.com/windows/apps/winui/winui3/)
- [Text rendering](https://docs.microsoft.com/windows/apps/project-reunion/dwritecore)
- [Localization/globalization](https://docs.microsoft.com/windows/apps/project-reunion/mrtcore/mrtcore-overview)

## Using Project Reunion 0.5 Preview with our ecosystem partners
Project Reunion 0.5 Preview is in the process of being integrated into several other important ecosystem technologies. To test out and follow that progress, check out some of our partner technologies below:

- [Windows Community Toolkit (Microsoft)](https://aka.ms/wct-winui3): The latest version of the WCT is working towards supporting 
Project Reunion 0.5 Preview in the near future, and provides tons of new controls and capabilities for use in your WinUI app. You can check out their open source repo [here](https://aka.ms/windowstoolkit).
- [Uno Platform](https://github.com/unoplatform): Developers using WinUI have an option to take WinUI-built applications to WebAssembly, Linux, macOS, Android and iOS via open-source project Uno Platform, and easily style them with Material and Fluent UI design systems for pixel-perfect experience everywhere. To set up your environment and get started with Uno Platform follow instructions laid out [here](https://platform.uno/docs/articles/getting-started-tutorial-1.html).
- [Telerik](https://www.telerik.com/winui): Telerik UI for WinUI is a UI component suite targeting WinUI 3 UWP apps currently released as Preview. The Telerik UI Preview includes several feature-rich controls including Ribbon, DataGrid, Charts, Gauges, Barcode, and more.
- [Syncfusion](https://www.syncfusion.com/winui-controls): The Syncfusion WinUI controls preview includes a large variety of high performance and modern controls for WinUI 3 UWP apps. The initial preview includes a suite of controls for data visualization, notifications, navigation, and content editing.
- [DevExpress](https://community.devexpress.com/blogs/wpf/archive/2020/06/11/devexpress-controls-for-winui-3-preview-are-now-available.aspx): The entire suite of DevExpress UWP controls is available in preview for WinUI 3 UWP apps. Some example controls include Scheduler, Charts, Navigation, DataGrid, and more
- [GrapeCity's service components](https://www.grapecity.com/componentone/net-service-component) allow users to integrate data from popular online data sources such as Dynamics, Salesforce, Quickbooks and Google Analytics, and optimize performance with data virtualization. GrapeCity also plans to bring their popular desktop UI controls to WinUI3 later this year.
- [Infragistics](https://www.infragistics.com/products/winui): Ultimate UI for WinUI brings business critical, high performing, and feature rich line of business controls to your apps that target any platform that runs Windows, including the new ARM64-based Surface Pro X. The Ultimate UI or WinUI Preview includes DataGrid, Charts (50+), Gauges, and more.

## What's Next
To keep up with progress being made on Project Reunion, please see our [feature roadmap](https://github.com/microsoft/ProjectReunion/blob/main/docs/roadmap.md) which gets updated regularly.

**Comment by jindal1979** [comment]
Is there plan to release final version of Reunion 0.5 which can be used for production ? Or production will be allowed from Reunion 1.0.

**Reply by andrewleader** [reply]
Hi @jindal1979, we're planning on releasing the final version of Project Reunion 0.5 by end of this month if all goes well. You **can** use the 0.5 GA release for production apps 👍 Note that 0.5 only supports packaged (MSIX) desktop apps, and the only features included in 0.5 are WinUI 3, MRT Core, and DWriteCore, and WinUI doesn't yet support incremental usage in your existing projects (it's best to start with a File->New WinUI 3 project and copy/paste).

**Reply by riverar** [reply]
> You **can** use the 0.5 GA release for production apps 👍

This is confusing. The announcement and version communicates "pre-release" / there be dragons / don't ship, but your comment says go for it. Which statement is true?

**Reply by andrewleader** [reply]
@riverar the 0.5 GA (general availability) release is different than the 0.5 preview release. Munish was asking about when the final version of 0.5 would be released and whether it would support production apps. Therefore, in the context of Munish's question (and replying directly to it), this answer is about 0.5 GA (even though this overall post is about the preview version which is not yet ready for production since it's still preview, we're releasing the final version soon).

**Reply by riverar** [reply]
> @riverar the 0.5 GA (general availability) release is different than the 0.5 preview release. [...] this answer is about 0.5 GA (even though this overall post is about the preview version which is not yet ready for production since it's still preview, we're releasing the final version soon).

Ah thanks. Very confusing because the docs/bits for this preview went live a few weeks ago. I guess those were 0.5-prerelease _candidates_ until now.
